element_interface/utils.py

```diff
@@ -140,31 +145,57 @@ def ingest_csv_to_table(
         )
         if verbose:
             insert_len = len(table) - prev_len
-            print(
+            logger.info(
                 f"\n---- Inserting {insert_len} entry(s) "
                 + f"into {table.table_name} ----"
             )
 
 
-def recursive_search(key: str, dictionary: dict) -> any:
-    """Return value for key in a nested dictionary
-
-    Search through a nested dictionary for a key and returns its value.  If there are
-    more than one key with the same name at different depths, the algorithm returns the
-    value of the least nested key.
+def str_to_bool(value) -> bool:
+    """Return whether the provided string represents true. Otherwise false.
 
     Args:
-        key (str): Key used to search through a nested dictionary
-        dictionary (dict): Nested dictionary
+        value (any): Any input
 
     Returns:
-        value (any): value of the input argument `key`
+        bool (bool): True if value in ("y", "yes", "t", "true", "on", "1")
+    """
+    # Due to distutils equivalent depreciation in 3.10
+    # Adopted from github.com/PostHog/posthog/blob/master/posthog/utils.py
+    if not value:
+        return False
+    return str(value).lower() in ("y", "yes", "t", "true", "on", "1")
+
+
+def write_csv(content: list, path: pathlib.PosixPath):
+    """General function for writing strings to lines in CSV
+
+    Args
+        content (list): list of strings, each as row of CSV
+        path (pathlib.PosixPath): where to wriite new CSV
     """
-    if key in dictionary:
-        return dictionary[key]
-    for value in dictionary.values():
-        if isinstance(value, dict):
-            a = recursive_search(key, value)
-            if a is not None:
-                return a
-    return None
+    with open(path, "w") as f:
+        for line in content:
+            f.write(line + "\n")
+
+
+class QuietStdOut:
+    """Context for quieting standard output, and setting datajoint loglevel to warning
+
+    Used in pytest functions to render cleaer output showing only pass/fail
+
+    Example:
+        with QuietStdOut:
+            table.delete(safemode=False)
+
+    """
+
+    def __enter__(self):
+        os.environ["DJ_LOG_LEVEL"] = "WARNING"
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self):  # , exc_type, exc_val, exc_tb):
```

Comment:
```suggestion
    def __exit__(self):
```

Reply:
Is there a specific reason to keep this commented code?

Reply:
No, this is a holdover from playing around with other functionality

Reply:
Oops - I just encountered a case when these values are needed. Using this as a context pass back a set of items. The __exit__ needs to be able to handle these. I'm pushing another commit that will change to accept `*args`